files/image_config/platform/rc.local

```diff
@@ -29,8 +39,60 @@ if [ ! -e /host/machine.conf ]; then
         done
     fi
 
+    # Extract the previous NOS's partition that contains the migration artifacts
+    set -- $(cat /proc/cmdline)
+    for x in "$@"; do
+        case "$x" in
+            nos-config-part=*)
+                nos_val="${x#nos-config-part=}"
+                ;;
+        esac
+    done
+
+    if [ -n "$nos_val" ]; then
+        nos_dev=$(findfs $nos_val)
+        if [ $? != 0 ]; then
+            log_migration "ERROR: nos_dev not found. Check grub parameters"
+        fi
+    else
+        log_migration "ERROR: nos_val not found. Check grub parameters"
+    fi
+
+    if [ -n "$nos_dev" ]; then
+        # Mount the previous NOS's partition
+        mkdir -p /mnt/nos_migration
+        mount $nos_dev /mnt/nos_migration
+        mkdir -p /etc/sonic/migration
+
+        # Copy relevant files
+        nos_migration_import /mnt/nos_migration/minigraph.xml /etc/sonic/migration
```

Comment:
We could import the entire bunch of files 1 shot - but this is just to ensure we know what we are importing.